bin/compile

```diff
@@ -79,19 +79,32 @@ heading() {
 APP_DIR=$(cd $(dirname "${BASH_SOURCE[0]}"); cd ..; pwd)
 BIN_DIR="$APP_DIR/vendor/bin"
 
-decrypt_output=$(echo $EJSON_PRIVATE_KEY | $BIN_DIR/ejson decrypt --key-from-stdin "$APP_DIR/$EJSON_FILE" 2>&1)
+exports=$(echo $EJSON_PRIVATE_KEY | \
+  $BIN_DIR/ejson decrypt --key-from-stdin "$APP_DIR/$EJSON_FILE" 2>&1 | \
+  $BIN_DIR/jq -r 'to_entries|map("export \(.key)=\(.value|tojson)")[]' 2>&1 | \
+  grep -v '^export _public_key=' | \
+  while IFS=$'\n' read -r line
+  do
+    echo $line
+  done)
+
 if [ $? -eq 0 ]; then
-  exports=$(echo $decrypt_output | $BIN_DIR/jq -r "to_entries|map(\"\(.key)=\(.value|tostring)\")|.[]" | grep -v '^_public_key=')
-  for env_var in $exports; do
-    export $env_var
-  done
+  eval "$exports"
 else
-  heading $decrypt_output
+  heading "Decryption failed"
+
+  # Don't exit 1 here to allow heroku ps:exec to work
```

Comment:
Can we get a little more context on the "why" here please?

Reply:
Updated with more context

Reply:
It might be more appropriate to define a function (`export_ejson_secrets` or similar) in this [`.profile.d` script](https://devcenter.heroku.com/articles/buildpack-api#profile-d-scripts) and require it to be called from the application's [`.profile` script](https://devcenter.heroku.com/articles/dynos#the-profile-file). Thoughts? cc @gstib @patrobinson

Reply:
I'm ok with that suggestion. I suspect that might work better for the Heroku <-> Docker stuff I'm doing as it would then work the same for either technique.